packages/dds/tree/src/test/feature-libraries/chunked-forest/chunkEncodingEndToEnd.ts

```diff
@@ -0,0 +1,159 @@
+		const treeSchema = new TreeStoredSchemaRepository(intoStoredSchema(jsonSequenceRootSchema));
+		const chunker = new Chunker(
+			treeSchema,
+			defaultSchemaPolicy,
+			Number.POSITIVE_INFINITY,
+			Number.POSITIVE_INFINITY,
+			defaultChunkPolicy.uniformChunkNodeCount,
+			tryShapeFromSchema,
+		);
+
+		const forest = buildChunkedForest(chunker);
+		const numberShape = new TreeShape(leaf.number.name, true, []);
+		const chunk = new UniformChunk(numberShape.withTopLevelLength(4), [1, 2, 3, 4]);
+		assert(!chunk.isShared());
+		const chunkCursor = chunk.cursor();
+
+		// This function is declared in the test to have access to the original uniform chunk for comparison.
+		function createFlexTree(editableForest: IEditableForest, schema: FlexTreeSchema): Context {
+			const changes: ModularChangeset[] = [];
+			const changeReceiver = (change: ModularChangeset) => {
+				assert(change.builds !== undefined);
+				// TODO: This chunk is actually a BasicChunk which was split from the original UniformChunk split up.
+				// This is expected to fail currently, but should eventually pass once we have the ability to insert UniformChunk.
+				const insertedChunk = change.builds.get(undefined)?.get(0 as ChangesetLocalId);
+				assert.equal(insertedChunk, chunk);
+				assert(chunk.isShared());
+				changes.push(change);
+			};
+			const dummyEditor = new DefaultEditBuilder(
+				new DefaultChangeFamily({ jsonValidator: noopValidator }),
+				changeReceiver,
+			);
+			return getTreeContext(
+				schema,
+				editableForest,
+				dummyEditor,
+				createMockNodeKeyManager(),
+				brand(nodeKeyFieldKey),
+			);
+		}
+
+		const flexTree = createFlexTree(forest, jsonSequenceRootSchema);
+		// This conversion is currently required for inserting a uniform chunk with a topLevelLength > 1.
+		const content: ITreeCursorSynchronous[] = prepareFieldCursorForInsert(chunkCursor);
+
+		flexTree.editor
+			.sequenceField({ parent: undefined, field: rootFieldKey })
+			.insert(0, content);
+	});
+
+	it(`summary values are correct, and shares reference with the original chunk.`, async () => {
+		const factory = new SharedTreeFactory({
+			jsonValidator: typeboxValidator,
+			forest: ForestType.Optimized,
+			summaryEncodeType: TreeCompressionStrategy.Compressed,
+		});
+		const provider = await TestTreeProvider.create(1, SummarizeType.onDemand, factory);
+		const tree = provider.trees[0];
+		const flexTree = tree.schematizeInternal({
+			allowedSchemaModifications: AllowedUpdateType.None,
+			schema: jsonSequenceRootSchema,
+			initialTree: [],
+		});
+
+		const numberShape = new TreeShape(leaf.number.name, true, []);
+		const chunk = new UniformChunk(numberShape.withTopLevelLength(4), [1, 2, 3, 4]);
+		assert(!chunk.isShared());
+
+		flexTree.editableTree.insertAt(0, chunk.cursor());
+		await provider.ensureSynchronized();
+
+		const options = {
+			jsonValidator: noopValidator,
```

Comment:
might as well use the typeboxValidator for extra validation here.

Reply:
Updated to use typeboxValidator